src/screens/send/components/address-book/address-book.tsx

```diff
@@ -0,0 +1,211 @@
+        const message = translate('Send.alertEditDescription');
+        const buttons = [
+            {
+                text: translate('App.labels.cancel'),
+                onPress: () => {
+                    /* console.log('Cancel Pressed')*/
+                },
+                type: 'cancel'
+            },
+            {
+                text: translate('App.labels.save'),
+                onPress: (inputValue: string) => {
+                    if (inputValue !== '') {
+                        const data: IContactState = {
+                            address: contact.address,
+                            blockchain: contact.blockchain,
+                            name: inputValue
+                        };
+                        this.closeCurrentOpenedSwipable();
+                        this.props.updateContactName(data);
+                    }
+                },
+                type: 'default'
+            }
+        ];
+        const type = 'plain-text';
+
+        Alert.prompt(title, message, buttons, type);
+    }
+
+    public renderLeftActions = (contact: IContactState) => {
+        const styles = this.props.styles;
+        return (
+            <View style={styles.leftActionsContainer}>
+                <TouchableOpacity
+                    style={styles.action}
+                    onPress={() => {
+                        confirm(translate('Send.deleteContact'), '')
+                            .then(() => {
+                                this.closeCurrentOpenedSwipable();
+                                this.props.deleteContact(contact);
+                            })
+                            .catch(() => {
+                                //
+                            });
+                    }}
+                >
+                    <Icon name="bin" size={32} style={styles.iconActionNegative} />
+                    <Text style={styles.textActionNegative}>{translate('Send.deleteContact')}</Text>
+                </TouchableOpacity>
+
+                <TouchableOpacity style={styles.action} onPress={() => this.onPressUpdate(contact)}>
+                    <Icon name="pencil" size={28} style={styles.iconActionPositive} />
+                    <Text style={styles.textActionPositive}>
+                        {translate('Send.editContactName')}
+                    </Text>
+                </TouchableOpacity>
+            </View>
+        );
+    };
+
+    public renderContact(contact: IContactState) {
+        const styles = this.props.styles;
+        const index = contact.address;
+
+        return (
+            <Swipeable
+                key={index}
+                ref={ref => {
+                    this.walletSwipableRef[index] = ref;
+                }}
+                renderLeftActions={() => this.renderLeftActions(contact)}
+                onSwipeableWillOpen={() => {
+                    if (
+                        index !== this.state.openedSwipeIndex &&
```

Comment:
since `openedSwipeIndex` doesn't affect rendering directly you can move it from state to class attribute. Every time you do a setState and something changes, you trigger a new render cycle and, in this case, you don't need that.